### PR TITLE
[addons] bye, bye CHelper_libXBMC_codec

### PR DESCRIPTION
**Pull request by AlwinEsch**
This remove the `CHelper_libXBMC_codec` and do the part from them direct on `CHelper_libXBMC_pvr`.

One question, makes it sense on end to have a function like `xbmc_codec_t GetCodecByName(const char* strCodecName)` as general for a independent add-on's (like Python)? In this case comes it on final to standard general calls.

### CODE REVIEW

File: Kodi.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -419,7 +419,6 @@
 		76AEFB371C8F79C100EF2EC0 /* AddonCallbacksAddon.cpp in Sources */ = {isa = PBXBuildFile; fileRef = EDED2E921C878EE0000F5E80 /* AddonCallbacksAddon.cpp */; };
 		76AEFB381C8F79C400EF2EC0 /* AddonCallbacksAudioDSP.cpp in Sources */ = {isa = PBXBuildFile; fileRef = EDED2E951C878EF8000F5E80 /* AddonCallbacksAudioDSP.cpp */; };
 		76AEFB391C8F79C700EF2EC0 /* AddonCallbacksAudioEngine.cpp in Sources */ = {isa = PBXBuildFile; fileRef = EDED2E8F1C878EB8000F5E80 /* AddonCallbacksAudioEngine.cpp */; };
-		76AEFB3A1C8F79CA00EF2EC0 /* AddonCallbacksCodec.cpp in Sources */ = {isa = PBXBuildFile; fileRef = EDED2E8C1C878E97000F5E80 /* AddonCallbacksCodec.cpp */; };
```
Comment (FernetMenta) [inline Kodi.xcodeproj/project.pbxproj]: ignore this this file, it is useless now

File: xbmc/addons/interfaces/PVR/AddonCallbacksPVR.cpp [hunk 2]
```diff
@@ -47,6 +51,67 @@ namespace KodiAPI
...
+
+private:
+  CCodecIds(void)
+  {
+    // get ids and names
+    AVCodec* codec = NULL;
+    xbmc_codec_t tmp;
+    while ((codec = av_codec_next(codec)))
+    {
+      if (av_codec_is_decoder(codec))
```
Comment (FernetMenta) [inline xbmc/addons/interfaces/PVR/AddonCallbacksPVR.cpp]: drop the dependency ffmpeg here, that makes no sense

Comment (AlwinEsch) [inline xbmc/addons/interfaces/PVR/AddonCallbacksPVR.cpp]: Is currently a bit hard to remove so long it is needed.
https://github.com/kodi-pvr/pvr.vdr.vnsi/blob/master/src/VNSIDemux.cpp#L304
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp#L689

Thought to remove it complete by the rework to multiple instance way and use inputstream on pvr.

Comment (FernetMenta) [inline xbmc/addons/interfaces/PVR/AddonCallbacksPVR.cpp]: get addon authors acquainted to more frequent breakage :)

Comment (AlwinEsch) [inline xbmc/addons/interfaces/PVR/AddonCallbacksPVR.cpp]: Not normal on PVR ;)

**Comment by AlwinEsch** [comment]
There only three pvr add-ons who use it, this are `pvr.mythtv`, `pvr.vdr.vnsi` and `pvr.hts`.

This are still on https://github.com/kodi-pvr only need the `kodi-agile` branch selected!

**Pull request merged**